File: SDKs/amazonaws.com/dms/2016-01-01/java/lib/src/main/java/openapisdk/models/shared/CreateEndpointMessage.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * CreateEndpointMessage
 * <p/>
**/
public class CreateEndpointMessage {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("CertificateArn")
    public String certificateArn;
    public CreateEndpointMessage withCertificateArn(String certificateArn) {
        this.certificateArn = certificateArn;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("DatabaseName")
    public String databaseName;
    public CreateEndpointMessage withDatabaseName(String databaseName) {
        this.databaseName = databaseName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("DmsTransferSettings")
    public DmsTransferSettings dmsTransferSettings;
    public CreateEndpointMessage withDmsTransferSettings(DmsTransferSettings dmsTransferSettings) {
        this.dmsTransferSettings = dmsTransferSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("DocDbSettings")
    public DocDbSettings docDbSettings;
    public CreateEndpointMessage withDocDbSettings(DocDbSettings docDbSettings) {
        this.docDbSettings = docDbSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("DynamoDbSettings")
    public DynamoDbSettings dynamoDbSettings;
    public CreateEndpointMessage withDynamoDbSettings(DynamoDbSettings dynamoDbSettings) {
        this.dynamoDbSettings = dynamoDbSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ElasticsearchSettings")
    public ElasticsearchSettings elasticsearchSettings;
    public CreateEndpointMessage withElasticsearchSettings(ElasticsearchSettings elasticsearchSettings) {
        this.elasticsearchSettings = elasticsearchSettings;
        return this;
    }
    @JsonProperty("EndpointIdentifier")
    public String endpointIdentifier;
    public CreateEndpointMessage withEndpointIdentifier(String endpointIdentifier) {
        this.endpointIdentifier = endpointIdentifier;
        return this;
    }
    @JsonProperty("EndpointType")
    public ReplicationEndpointTypeValueEnum endpointType;
    public CreateEndpointMessage withEndpointType(ReplicationEndpointTypeValueEnum endpointType) {
        this.endpointType = endpointType;
        return this;
    }
    @JsonProperty("EngineName")
    public String engineName;
    public CreateEndpointMessage withEngineName(String engineName) {
        this.engineName = engineName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ExternalTableDefinition")
    public String externalTableDefinition;
    public CreateEndpointMessage withExternalTableDefinition(String externalTableDefinition) {
        this.externalTableDefinition = externalTableDefinition;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ExtraConnectionAttributes")
    public String extraConnectionAttributes;
    public CreateEndpointMessage withExtraConnectionAttributes(String extraConnectionAttributes) {
        this.extraConnectionAttributes = extraConnectionAttributes;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("IBMDb2Settings")
    public IbmDb2Settings ibmDb2Settings;
    public CreateEndpointMessage withIbmDb2Settings(IbmDb2Settings ibmDb2Settings) {
        this.ibmDb2Settings = ibmDb2Settings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("KafkaSettings")
    public KafkaSettings kafkaSettings;
    public CreateEndpointMessage withKafkaSettings(KafkaSettings kafkaSettings) {
        this.kafkaSettings = kafkaSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("KinesisSettings")
    public KinesisSettings kinesisSettings;
    public CreateEndpointMessage withKinesisSettings(KinesisSettings kinesisSettings) {
        this.kinesisSettings = kinesisSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("KmsKeyId")
    public String kmsKeyId;
    public CreateEndpointMessage withKmsKeyId(String kmsKeyId) {
        this.kmsKeyId = kmsKeyId;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("MicrosoftSQLServerSettings")
    public MicrosoftSqlServerSettings microsoftSQLServerSettings;
    public CreateEndpointMessage withMicrosoftSqlServerSettings(MicrosoftSqlServerSettings microsoftSQLServerSettings) {
        this.microsoftSQLServerSettings = microsoftSQLServerSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("MongoDbSettings")
    public MongoDbSettings mongoDbSettings;
    public CreateEndpointMessage withMongoDbSettings(MongoDbSettings mongoDbSettings) {
        this.mongoDbSettings = mongoDbSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("MySQLSettings")
    public MySqlSettings mySQLSettings;
    public CreateEndpointMessage withMySqlSettings(MySqlSettings mySQLSettings) {
        this.mySQLSettings = mySQLSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("NeptuneSettings")
    public NeptuneSettings neptuneSettings;
    public CreateEndpointMessage withNeptuneSettings(NeptuneSettings neptuneSettings) {
        this.neptuneSettings = neptuneSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("OracleSettings")
    public OracleSettings oracleSettings;
    public CreateEndpointMessage withOracleSettings(OracleSettings oracleSettings) {
        this.oracleSettings = oracleSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Password")
    public String password;
    public CreateEndpointMessage withPassword(String password) {
        this.password = password;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Port")
    public Long port;
    public CreateEndpointMessage withPort(Long port) {
        this.port = port;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("PostgreSQLSettings")
    public PostgreSqlSettings postgreSQLSettings;
    public CreateEndpointMessage withPostgreSqlSettings(PostgreSqlSettings postgreSQLSettings) {
        this.postgreSQLSettings = postgreSQLSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("RedisSettings")
    public RedisSettings redisSettings;
    public CreateEndpointMessage withRedisSettings(RedisSettings redisSettings) {
        this.redisSettings = redisSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("RedshiftSettings")
    public RedshiftSettings redshiftSettings;
    public CreateEndpointMessage withRedshiftSettings(RedshiftSettings redshiftSettings) {
        this.redshiftSettings = redshiftSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ResourceIdentifier")
    public String resourceIdentifier;
    public CreateEndpointMessage withResourceIdentifier(String resourceIdentifier) {
        this.resourceIdentifier = resourceIdentifier;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("S3Settings")
    public S3Settings s3Settings;
    public CreateEndpointMessage withS3Settings(S3Settings s3Settings) {
        this.s3Settings = s3Settings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ServerName")
    public String serverName;
    public CreateEndpointMessage withServerName(String serverName) {
        this.serverName = serverName;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ServiceAccessRoleArn")
    public String serviceAccessRoleArn;
    public CreateEndpointMessage withServiceAccessRoleArn(String serviceAccessRoleArn) {
        this.serviceAccessRoleArn = serviceAccessRoleArn;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("SslMode")
    public DmsSslModeValueEnum sslMode;
    public CreateEndpointMessage withSslMode(DmsSslModeValueEnum sslMode) {
        this.sslMode = sslMode;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("SybaseSettings")
    public SybaseSettings sybaseSettings;
    public CreateEndpointMessage withSybaseSettings(SybaseSettings sybaseSettings) {
        this.sybaseSettings = sybaseSettings;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Tags")
    public Tag[] tags;
    public CreateEndpointMessage withTags(Tag[] tags) {
        this.tags = tags;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Username")
    public String username;
    public CreateEndpointMessage withUsername(String username) {
        this.username = username;
        return this;
    }
}

File: SDKs/amazonaws.com/dms/2016-01-01/python/src/sdk/models/shared/createendpointmessage.py
import dataclasses
from typing import Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import dmstransfersettings as shared_dmstransfersettings
from ..shared import docdbsettings as shared_docdbsettings
from ..shared import dynamodbsettings as shared_dynamodbsettings
from ..shared import elasticsearchsettings as shared_elasticsearchsettings
from ..shared import replicationendpointtypevalue_enum as shared_replicationendpointtypevalue_enum
from ..shared import ibmdb2settings as shared_ibmdb2settings
from ..shared import kafkasettings as shared_kafkasettings
from ..shared import kinesissettings as shared_kinesissettings
from ..shared import microsoftsqlserversettings as shared_microsoftsqlserversettings
from ..shared import mongodbsettings as shared_mongodbsettings
from ..shared import mysqlsettings as shared_mysqlsettings
from ..shared import neptunesettings as shared_neptunesettings
from ..shared import oraclesettings as shared_oraclesettings
from ..shared import postgresqlsettings as shared_postgresqlsettings
from ..shared import redissettings as shared_redissettings
from ..shared import redshiftsettings as shared_redshiftsettings
from ..shared import s3settings as shared_s3settings
from ..shared import dmssslmodevalue_enum as shared_dmssslmodevalue_enum
from ..shared import sybasesettings as shared_sybasesettings
from ..shared import tag as shared_tag


@dataclass_json
@dataclasses.dataclass
class CreateEndpointMessage:
    r"""CreateEndpointMessage
    <p/>
    """
    
    endpoint_identifier: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('EndpointIdentifier') }})
    endpoint_type: shared_replicationendpointtypevalue_enum.ReplicationEndpointTypeValueEnum = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('EndpointType') }})
    engine_name: str = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('EngineName') }})
    certificate_arn: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('CertificateArn') }})
    database_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DatabaseName') }})
    dms_transfer_settings: Optional[shared_dmstransfersettings.DmsTransferSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DmsTransferSettings') }})
    doc_db_settings: Optional[shared_docdbsettings.DocDbSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DocDbSettings') }})
    dynamo_db_settings: Optional[shared_dynamodbsettings.DynamoDbSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('DynamoDbSettings') }})
    elasticsearch_settings: Optional[shared_elasticsearchsettings.ElasticsearchSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ElasticsearchSettings') }})
    external_table_definition: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ExternalTableDefinition') }})
    extra_connection_attributes: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ExtraConnectionAttributes') }})
    ibm_db2_settings: Optional[shared_ibmdb2settings.IbmDb2Settings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('IBMDb2Settings') }})
    kafka_settings: Optional[shared_kafkasettings.KafkaSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('KafkaSettings') }})
    kinesis_settings: Optional[shared_kinesissettings.KinesisSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('KinesisSettings') }})
    kms_key_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('KmsKeyId') }})
    microsoft_sql_server_settings: Optional[shared_microsoftsqlserversettings.MicrosoftSQLServerSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MicrosoftSQLServerSettings') }})
    mongo_db_settings: Optional[shared_mongodbsettings.MongoDbSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MongoDbSettings') }})
    my_sql_settings: Optional[shared_mysqlsettings.MySQLSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('MySQLSettings') }})
    neptune_settings: Optional[shared_neptunesettings.NeptuneSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('NeptuneSettings') }})
    oracle_settings: Optional[shared_oraclesettings.OracleSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('OracleSettings') }})
    password: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Password') }})
    port: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Port') }})
    postgre_sql_settings: Optional[shared_postgresqlsettings.PostgreSQLSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('PostgreSQLSettings') }})
    redis_settings: Optional[shared_redissettings.RedisSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('RedisSettings') }})
    redshift_settings: Optional[shared_redshiftsettings.RedshiftSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('RedshiftSettings') }})
    resource_identifier: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ResourceIdentifier') }})
    s3_settings: Optional[shared_s3settings.S3Settings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('S3Settings') }})
    server_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ServerName') }})
    service_access_role_arn: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ServiceAccessRoleArn') }})
    ssl_mode: Optional[shared_dmssslmodevalue_enum.DmsSslModeValueEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('SslMode') }})
    sybase_settings: Optional[shared_sybasesettings.SybaseSettings] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('SybaseSettings') }})
    tags: Optional[list[shared_tag.Tag]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Tags') }})
    username: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Username') }})
    
